test_framework/scripts/longhorn-setup.sh

```diff
@@ -290,16 +337,29 @@ main(){
 	create_aws_secret
 	set -x
 	install_csi_snapshotter_crds
-	generate_longhorn_yaml_manifest "${TF_VAR_tf_workspace}"
 
-	if [[ "${LONGHORN_UPGRADE_TEST}" == true || "${LONGHORN_UPGRADE_TEST}" == True ]]; then
-		install_longhorn_stable
-		run_longhorn_upgrade_test ${WORKSPACE}
-		run_longhorn_tests ${WORKSPACE}
-	else
-		install_longhorn_master "${TF_VAR_tf_workspace}/longhorn.yaml"
-		run_longhorn_tests ${WORKSPACE}
-	fi
+  if [[ "${AIR_GAP_INSTALLATION}" == true ]]; then
+    create_registry_secret
+    if [[ "${LONGHORN_INSTALL_METHOD}" == "manifest-file" ]]; then
+      get_longhorn_manifest
+      customize_longhorn_manifest_for_airgap
```

Comment:
We need to ensure to set the setting (https://longhorn.io/docs/1.3.0/references/settings/#registry-secret) as well, otherwise the following pods created by longhorn-manager will have errors. (like share manager pods, backing image manager pods, etc)

Reply:
Refined

Reply:
**For installation from manifest file**, setting `registry-secret` in `default-setting.yaml` works as expected. If the value is set, longhorn regression test can pass as normal, otherwise there would be lots of ImagePullBackOff errors during the test.

**But for installation from helm chart**, even though we set the `defaultSettings.registrySecret` value in `chart/values.yaml `according to the [documentation](https://longhorn.io/docs/1.3.0/advanced-resources/deploy/airgap/#use-default-image-name), the value would not be set in `default-setting.yaml` since it's been removed in [this commit](https://github.com/longhorn/longhorn/commit/b79b914967a84d66c5bb141e8056cd638d52ace2#diff-b4292a87e40a1ece0e015a805cf4ccc28516e54f3bc3e5a36e03526f8179004b), so longhorn-manager would still have errors during the test.

Reply:
Thanks, @yangchiu found this. For the chart install/upgrade, the setting will be implicitly updated as per the secret from the manage daemonset as below, so we don't need to update that setting separately. 

https://github.com/longhorn/longhorn-manager/blob/ddf1b8dfe8f3ccceeeafff2909d9c5763bf33122/app/daemon.go#L155